internal/timeseries/parquet_collection.go

```diff
@@ -0,0 +1,73 @@
+package timeseries
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/innosat-mats/rac-extract-payload/internal/common"
+)
+
+// ParquetFactory is a function that creates ParquetWriters
+type ParquetFactory func(pkg *common.DataRecord) (ParquetWriter, error)
+
+// ParquetCollection holds all active ParquetWriters
+type ParquetCollection struct {
+	streams map[string]ParquetWriter
+	factory ParquetFactory
+}
+
+// NewParquetCollection returns a novel ready to use ParquetCollection
+func NewParquetCollection(factory ParquetFactory) ParquetCollection {
+	return ParquetCollection{factory: factory, streams: make(map[string]ParquetWriter)}
+}
+
+// ParquetName returns the whole name of the parquet, including partitioning prefix
+func ParquetName(pkg *common.DataRecord) string {
+	tmTime := pkg.TMHeader.Time(time.Time{})
+	prefix := filepath.Join(
+		fmt.Sprintf("%v", tmTime.Year()),
+		fmt.Sprintf("%v", int(tmTime.Month())),
+		fmt.Sprintf("%v", tmTime.Day()),
+	)
+	baseName := filepath.Base(pkg.Origin.Name)

```

Comment:
Is `pkg.Origin.Name` the `.rac` file name? If not, can it be the same for multiple `pkg`?

Reply:
It is the name of the rac-file.